In [1]:
import dask as dd
import numpy as np
import pandas as pd
from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score
from sklearn.tree import DecisionTreeRegressor
import pickle
import numpy as np
import inspect
import os

In [4]:
os.chdir("/Users/richardtran/downloads")
df = pd.read_csv('exportdf.csv')
df['price'] = df['price'].str.replace('$', '')
df['price'] = df['price'].str.replace(',', '')
df["price"] = pd.to_numeric(df["price"])

/Users/richardtran/opt/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3063: DtypeWarning: Columns (61,62,94,95) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [5]:
df2 = df[['ntaname','room_type','property_type','price']]

def Encode(df2):
    for column in df2.columns[df2.columns.isin(['ntaname', 'room_type','property_type'])]:
        df2[column] = df2[column].factorize()[0]
    return df2

df3 = Encode(df2.copy())

In [6]:
x = df3.iloc[:,[0,1,2]]
y = df3['price']

x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=.1,random_state=1)
DTree=DecisionTreeRegressor(min_samples_leaf=.0001)
DTree.fit(x_train,y_train)
y_predict=DTree.predict(x_test)
r2_score(y_test,y_predict)


0.39069913854927363

In [12]:
# Saving model to disk
pickle.dump(DTree, open('nyc_model3.pkl','wb'))

# Loading model to compare the results
model = pickle.load(open('nyc_model3.pkl','rb'))
print(model.predict([[1, 1, 1]]))


[75.18705036]


In [148]:
DTree.predict([[1,1,1]])

array([75.18705036])

In [11]:
conda list scikit-learn

# packages in environment at /Users/richardtran/opt/anaconda3:
#
# Name                    Version                   Build  Channel
scikit-learn              0.22.2.post1             pypi_0    pypi

Note: you may need to restart the kernel to use updated packages.


In [79]:
df3=df3.rename(columns={"ntaname": "nta_id", "room_type": "room_type_id","property_type":"property_type_id","price":"Price"})
df4=pd.concat([df3, df2.reindex(df2.index)], axis=1)
df4.drop('price', inplace=True, axis=1)
df4.head(100)
#apt = 0


nta_id  room_type_id  property_type_id  Price                ntaname  \
0       0             0                 0  225.0  Midtown-Midtown South   
1       1             0                 1   89.0           Clinton Hill   
2       2             0                 0  200.0   Murray Hill-Kips Bay   
3       1             1                 0   60.0           Clinton Hill   
4       3             1                 0   79.0                Clinton   

         room_type property_type  
0  Entire home/apt     Apartment  
1  Entire home/apt   Guest suite  
2  Entire home/apt     Apartment  
3     Private room     Apartment  
4     Private room     Apartment

Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - dataframe

Current channels:

  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.



Note: you may need to restart the kernel to use updated packages.


In [121]:
df4.groupby('ntaname').count()['nta_id'].nlargest(30)

ntaname
North Side-South Side                         2285
Bushwick South                                1796
Clinton                                       1742
Stuyvesant Heights                            1567
Crown Heights North                           1558
Bedford                                       1476
East Village                                  1435
Hudson Yards-Chelsea-Flatiron-Union Square    1412
West Village                                  1230
Upper West Side                               1201
East Williamsburg                             1190
Bushwick North                                1160
Midtown-Midtown South                         1113
SoHo-TriBeCa-Civic Center-Little Italy        1024
Greenpoint                                    1013
Park Slope-Gowanus                             966
Central Harlem North-Polo Grounds              923
Chinatown                                      918
Clinton Hill                                   910
Lower East Side        

In [142]:
df4[df4.ntaname == 'Stapleton-Rosebank']

nta_id  room_type_id  property_type_id  Price             ntaname  \
561        77             0                 0   75.0  Stapleton-Rosebank   
812        77             1                 0   59.0  Stapleton-Rosebank   
905        77             1                 6   97.0  Stapleton-Rosebank   
1339       77             1                 6   71.0  Stapleton-Rosebank   
2674       77             1                16  100.0  Stapleton-Rosebank   
...       ...           ...               ...    ...                 ...   
47414      77             1                 6   65.0  Stapleton-Rosebank   
49272      77             1                 6   59.0  Stapleton-Rosebank   
49275      77             1                 6  199.0  Stapleton-Rosebank   
49449      77             0                 6   80.0  Stapleton-Rosebank   
49595      77             0                 0   75.0  Stapleton-Rosebank   

             room_type property_type  
561    Entire home/apt     Apartment  
812       Private room     Apartment  
905       Private room         House  
1339      Private room         House  
2674      Private room   Earth house  
...                ...           ...  
47414     Private room         House  
49272     Private room         House  
49275     Private room         House  
49449  Entire home/apt         House  
49595  Entire home/apt     Apartment  

[65 rows x 7 columns]

property - 
apt- 0
house - 6
condo - 5
Townhouse - 2

Entire home/apt - 0
private room - 1

Brooklyn   8 North Side-South Side    2285 
           26 Bushwick South           1796
           13 Stuyvesant Heights       1567
           11 Crown Heights North      1558
           42 Bedford                  1476
           19 East Williamsburg        1190
Manhattan  3 Clinton                                       1742
           16 East Village                                  1435
           10 Hudson Yards-Chelsea-Flatiron-Union Square    1412
           7 West Village                                  1230
           22 Upper West Side                               1201
           0 Midtown-Midtown South                         1113
Queens     54 Astoria                                 801
           25 Hunters Point-Sunnyside-West Maspeth    589
           45 Ridgewood                               480
           50 Steinway                                324
           82 Jackson Heights                         279
Bronx      46 Mott Haven-Port Morris      113
           118 Williamsbridge-Olinville     67
           Allerton-Pelham Gardens      59
           Van Cortlandt Village        57
           West Concourse               54
           Woodlawn-Wakefield           53
Staten Island  43 West New Brighton-New Brighton-St. George    79
               77 Stapleton-Rosebank                           65
               New Brighton-Silver Lake                     39
               Grasmere-Arrochar-Ft. Wadsworth              30
               Grymes Hill-Clifton-Fox Hills                27
               New Dorp-Midland Beach                       25